controllers/argocd/openshift/openshift.go

```diff
@@ -92,12 +92,37 @@ func ReconcilerHook(cr *argoprojv1alpha1.ArgoCD, v interface{}, hint string) err
 			}
 			policyRules := getPolicyRuleForApplicationController()
 			policyRules = append(policyRules, clusterRole.Rules...)
+			if !argoExists(policyRules) {
+				policyRules = append(policyRules, getPolicyRuleForNonOlmArgoCDApplicationController()...)
+			}
 			o.Rules = policyRules
 		}
 	}
 	return nil
 }
 
+// checks if argoproj.io rules exist in application controller role, if not return false
+func argoExists(policies []rbacv1.PolicyRule) (result bool) {
```

Comment:
please make the name more descriptive. This might also mean the ArgoCD instance exists.